drivers/hmis/app/models/hmis/ce/match/candidate_pool.rb

```diff
@@ -36,6 +37,10 @@ class CandidatePool < GrdaWarehouseBase
 
     attr_readonly :requirement_expression, :priority_expression
 
+    # Clean up the link from soft-deleted associations before deleting the pool. This avoids
+    # foreign key constraint violations
+    before_destroy :nullify_deleted_associations
```

Comment:
The alternative is to add soft-deletion to candidate pool. Not sure which is better. I guess I'd skip it since we keep orphan pools around for a while before we delete them, so we probably don't need long-term tracking on deleted pools ?